spring-cloud-sleuth-core/src/test/java/org/springframework/cloud/sleuth/instrument/web/SkipPatternProviderConfigTest.java

```diff
@@ -301,29 +293,13 @@ private SingleSkipPattern bar() {
 		return () -> Optional.of(Pattern.compile("bar"));
 	}
 
-	private ExposableWebEndpoint createEndpoint(final String name) {
-		return new ExposableWebEndpoint() {
-
-			@Override
-			public String getRootPath() {
-				return name;
-			}
+	@Configuration
+	@ImportAutoConfiguration({ DispatcherServletAutoConfiguration.class,
```

Comment:
Why don't we just add all these auto-configuration classes to the WebApplicationContextRunner? Then wouldn't the custom configuration class be unnecessary altogether?

Edit: I tried and it indeed doesn't work, but I don't understand why it is different.

Reply:
This is a good question, since I tried this first, and it didn't work out as well. There are few reasons (I think), since some tests do checks when there are no web endpoints. Another thing is `@EnableConfigurationProperties` which triggers the conditional match on `ServerProperties` bean (otherwise it is not present).

Reply:
@shakuzen I cut it to a bare minimum, do you think we could keep it? I didn't figure out the other way to ``ServerProperties`, might be part of another configuration.